astropy/units/quantity.py

```diff
@@ -156,6 +156,48 @@ def _construct_from_dict(self, map):
         value = map.pop('value')
         return self._parent_cls(value, **map)
 
+    def empty_like(self, cols, length, metadata_conflicts='warn', name=None):
```

Comment:
I very much like the concept, but am not quite sold on the name, as `np.empty_like` works differently. Maybe `shaped_like`? Or `new_like`? (not a big deal...)